NDTensors/test/NDTensorsTestUtils/device_list.jl

```diff
@@ -23,6 +23,10 @@ if "cutensor" in ARGS || "all" in ARGS
   Pkg.add("cuTENSOR")
   using CUDA, cuTENSOR
 end
+if "jlarrays" in ARGS || "all" in ARGS
+  Pkg.add("JLArrays")
+  using JLArrays
+end
```

Comment:
I don't think there is a reason to not run the tests for JLArrays since they run on CPU anyway. So I think it should just be added as a normal test dependency, and inserted into the device list by default.

Reply:
So I am seeing an issue with the GPU tests when I have `JLArrays` is always in the `Project.toml`. To fix the problem I `isempty(ARGS) || "base" in ARGS` so this would run into a problem if the test args are `["base","metal"]` but right now we aren't testing in that way. Though if you would prefer, I can just include `JLArrays` if no GPUs are being tested, i.e. only allow `JLArrays` if `isempty(ARGS)`. Let me know, thanks!

Reply:
What's the issue you see? It would be best to try to address that issue and have `JLArrays` as a dependency in `Project.toml` and run by default in the tests as we planned, so I would prefer to try to aim for that rather than work around an issue with a more convoluted solution.

Reply:
The issue is that GPUArrays does not compile during `Pkg.test`
```
ERROR: LoadError: UndefVarError: `SimplifyCFGPassOptions` not defined
Stacktrace:
 [1] top-level scope
   @ ~/.julia/packages/GPUCompiler/nWT2N/src/optim.jl:57
 [2] include(mod::Module, _path::String)
   @ Base ./Base.jl:495
 [3] include(x::String)
   @ GPUCompiler ~/.julia/packages/GPUCompiler/nWT2N/src/GPUCompiler.jl:1
 [4] top-level scope
   @ ~/.julia/packages/GPUCompiler/nWT2N/src/GPUCompiler.jl:37
 [5] include
   @ Base ./Base.jl:495 [inlined]
 [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::String)
   @ Base ./loading.jl:2216
 [7] top-level scope
   @ stdin:3
in expression starting at /home/jenkins/workspace/ITensor_ITensors.jl_PR-1508@tmp/.julia/packages/GPUCompiler/nWT2N/src/optim.jl:56
in expression starting at /home/jenkins/workspace/ITensor_ITensors.jl_PR-1508@tmp/.julia/packages/GPUCompiler/nWT2N/src/GPUCompiler.jl:1
in expression starting at stdin:3
ERROR: LoadError: Failed to precompile GPUCompiler [61eb1bfa-7361-4325-ad38-22787b887f55] to "/home/jenkins/workspace/ITensor_ITensors.jl_PR-1508@tmp/.julia/compiled/v1.10/GPUCompiler/jl_xlhZZG".
```
However, I think I was able to fix the problem by moving `using JLArrays` to after the GPU `using` statements.